pkg/ingester/ingester.go

```diff
@@ -843,7 +843,16 @@ func (i *Ingester) Label(ctx context.Context, req *logproto.LabelRequest) (*logp
 	if err != nil {
 		return nil, err
 	}
-	resp, err := instance.Label(ctx, req)
+
+	var matchers []*labels.Matcher
+	if req.Query != "" {
```

Comment:
Looks good I'm first sight t I'm wondering if it should be called `matchers`. I'm also widening if we have a different way to serialize them. I'll check.

Reply:
Well, here is your answer for me

> However I hesitated to do so as per the LokiLabelNamesRequest implementing the queryrangebase/Request interface that in turns includes a Query() method.

Could you add this as a comment to the proto?